_posts/2022-03-09-#19602.md

```diff
@@ -10,15 +10,60 @@ status: upcoming
+
+* [PR #16528](https://github.com/bitcoin/bitcoin/pull/16528) introduced Descriptor wallets, an
+  entirely new type of wallet which uses [output script descriptors](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md)
+  to manage keys and scripts in a wallet. This was implemented by introducing a
+  `DescriptorScriptPubKeyMan`.
+
+* The old type of wallets are referred to as Legacy wallets and they are slowly being deprecated
+  and removed. However there needs to be a way to turn a Legacy wallet into a Descriptor wallet
+  so that users do not find themselves unable to use their wallet.
+
+* In addition to changing how keys and scripts are tracked, Descriptor wallets also redefine the
+  concept (and implementation) of `IsMine`. Instead of allowing both private keys and watch-only addresses in a single
+  wallet, Descriptor wallets can either always have private keys, or never have private keys.
+
+* This PR implements an RPC which migrates a Legacy wallet into a descriptor wallet by computing
+  descriptors for everything the legacy wallet would have considered to belong to it. These
+  descriptors are subsequently added to a newly created descriptor wallet with the same name as
+  the original Legacy wallet.
+
+* Legacy wallets determine whether a script belongs to it by checking to see if it would be able
+  to sign for the script, or the script is explicitly being watched. Descriptor wallets determine
+  this by comparing the script to the set of scripts computed from the stored descriptors.
+
+* This PR is a step in the [roadmap](https://github.com/bitcoin/bitcoin/issues/20160) for the
+  eventual removal of the Legacy wallet and the Berkeley DB dependency.
+
 ## Questions
+
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
 
+2. Try creating a Legacy wallet and then migrating it using `migratewallet`. How can you tell if the migration was successful?
+
+   * A Legacy wallet can be created with `bitcoin-cli -named createwallet wallet_name=<name> descriptors=false`
+
+3. In your own words, briefly describe the migration process.
+
+4. What kind of scripts will a Legacy wallet consider to belong to it for a single key
+   (Hint: Look at [`LegacyScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L215))?
+
+5. Under what circumstances will a Legacy wallet watch for a multisig?
+
+6. Home come `LegacyScriptPubKeyMan::GetAllScriptPubKeys()` is needed
+   (Hint: Consider how [`DescriptorScriptPubKeyMan::IsMine`](https://github.com/bitcoin/bitcoin/blob/267917f5632a99bb51fc3fe516d8308e79d31ed1/src/wallet/scriptpubkeyman.cpp#L1705) works))?
+
+7. How come the HD seed is not ignored when looking at all of the keys in a Legacy wallet?
+
+8. What happens to watch-only things that were in the Legacy wallet?
```

Comment:
Instead of "things," maybe specify the objects?

Reply:
Done